### PR TITLE
Support to ignore a single test

### PR DESCRIPTION
**Pull request by eliasson**
Make it possible to ignore a single test.

Initial idea was to use a prefix like `ignore "it should..." in:` but I could not get the moons in orbit. So the current syntax is ScalaTest-like `"it should..." ignore:`

Next step for this feature would be to ignore an entire context, but this can go in as a separate PR.

image

### CODE REVIEW

File: src/main/scala/intent/core/internal.scala [hunk 1]
```diff
@@ -13,6 +13,10 @@ trait TestSupport extends FormatterGivens with EqGivens with ExpectGivens
 trait IntentStructure:
   private[intent] def allTestCases: Seq[ITestCase]
 
+  case class IgnoredTestCase(nameParts: Seq[String]) extends ITestCase:
```
Comment (provegard) [inline src/main/scala/intent/core/internal.scala]: Does this really need to be nested below `IntentStructure`?

File: src/test/scala/intent/runner/TestSuiteRunnerTest.scala [hunk 2]
```diff
@@ -49,11 +49,26 @@ class TestSuiteRunnerTest extends TestSuite with State[TestSuiteTestCase]:
...
+      "report that 1 test was ignored" in:
+        state =>
+          val possible = Await.result(state.runAll(), 5 seconds)
+          possible match
+            case Left(_) => fail("Unexpected Left")
+            case Right(result) => expect(result.ignored).toEqual(1)
+
       "with a registered event subscriber" using (_.copy()) to : // TODO: can we use identity here?
-        "should publish 3 events" in:
+        "should publish € events" in:
```
Comment (provegard) [inline src/test/scala/intent/runner/TestSuiteRunnerTest.scala]:
```suggestion
        "should publish 4 events" in:
```

Money money ;-)

File: src/test/scala/intent/runner/TestSuiteRunnerTest.scala [hunk 3]
```diff
@@ -49,11 +49,26 @@ class TestSuiteRunnerTest extends TestSuite with State[TestSuiteTestCase]:
             case Left(_) => fail("Unexpected Left")
             case Right(result) => expect(result.errors).toEqual(1)
 
+      "report that 1 test was ignored" in:
+        state =>
+          val possible = Await.result(state.runAll(), 5 seconds)
```
Comment (provegard) [inline src/test/scala/intent/runner/TestSuiteRunnerTest.scala]: We can use `whenComplete` now :-)

File: src/main/scala/intent/core/internal.scala (hunk 1 above)
Comment (eliasson) [reply src/main/scala/intent/core/internal.scala]: No reason, I'll extract it

**Pull request merged**